File: hw/ramos/u2/hw19Inheritance/Python/model/pig.py
from model.farmAnimal import FarmAnimal

class Pig(FarmAnimal):
    def __init__(self, foodPerDay, isReadyForSale, id, breed, bornOn, gender, isAbleToReproduce, weight, cage):
        super().__init__(id, breed, bornOn, gender, isAbleToReproduce, weight, cage)
        self.foodPerDay = foodPerDay
        self.isReadyForSale = isReadyForSale

    def feed(self, amount):
        self.foodPerDay += amount

    def __str__(self):
        return (
            f"-- Pig --\n"
            f"Food per Day: {self.foodPerDay} kg\n"
            f"Ready for Sale: {self.isReadyForSale}\n"
            f"{super().__str__()}\n"
            f"------------------------------"
        )

File: hw/santorum/u2/hw19Inheritance/JavaS/model/Pig.js
import { FarmAnimal } from "./FarmAnimal.js";

export class Pig extends FarmAnimal {
    constructor(foodPerDay, isReadyForSale, id, breed, bornOn, gender, isAbleToReproduce, weight, cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.foodPerDay = foodPerDay;
        this.isReadyForSale = isReadyForSale;
    }

    feed(amount) {
        this.foodPerDay += amount;
    }

    toString() {
        return `
=== PIG ===
Food per Day: ${this.foodPerDay} kg
Ready for Sale: ${this.isReadyForSale}
${super.toString()}
============================`;
    }
}
